whisper/src/bin/whisper-diff.rs

```diff
@@ -0,0 +1,119 @@
+#[macro_use]
+extern crate structopt;
+#[macro_use]
+extern crate serde_derive;
+extern crate failure;
+extern crate serde_json;
+extern crate whisper;
+
+use failure::Error;
+use std::path::PathBuf;
+use std::process::exit;
+use std::time::{SystemTime, UNIX_EPOCH};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "whisper-diff")]
+struct Args {
+    /// Path to data file
+    #[structopt(name = "path_a", parse(from_os_str))]
+    path_a: PathBuf,
+
+    /// Path to data file
+    #[structopt(name = "path_b", parse(from_os_str))]
+    path_b: PathBuf,
+
+    /// Show summary of differences
+    #[structopt(long = "summary")]
+    summary: bool,
+
+    /// Skip comparison if either value is undefined
+    #[structopt(long = "ignore-empty")]
+    ignore_empty: bool,
+
+    /// Print output in simple columns
+    #[structopt(long = "columns")]
+    columns: bool,
+
+    /// Do not print column headers
+    #[structopt(long = "no-headers")]
+    no_headers: bool,
+
+    /// Unix epoch time of the end of your requested
+    #[structopt(long = "until")]
+    until: Option<u32>,
+
+    /// Output results in JSON form
+    #[structopt(long = "json")]
+    json: bool,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct DiffArchiveContainer {
+    #[serde(flatten)]
+    inner: whisper::DiffArchive,
+    points: usize,
+}
+
+impl From<whisper::DiffArchive> for DiffArchiveContainer {
+    fn from(w: whisper::DiffArchive) -> DiffArchiveContainer {
+        DiffArchiveContainer {
+            points: w.diffs.len(),
+            inner: w,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct DiffArchiveSummary {
+    archive: u32,
+    total: u32,
+    points: u32,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct DiffArchiveJson {
+    path_a: String,
+    path_b: String,
+    archives: Vec<DiffArchiveContainer>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct DiffArchiveSummaryJson {
+    path_a: PathBuf,
+    path_b: PathBuf,
+    archives: Vec<DiffArchiveContainer>,
+}
+
+fn main() -> Result<(), Error> {
+    let args = Args::from_args();
+
+    for filename in &[&args.path_a, &args.path_b] {
+        if !filename.exists() {
+            eprintln!(
+                "[ERROR] File \"{}\" does not exist!",
+                filename.to_str().unwrap()
+            );
+            exit(1);
+        }
+    }
+
+    let now = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs() as u32;
```

Comment:
mb we should have int func to obtain seconds since epoch? because it used in several cli's and benches/tests

Reply:
What is a problem with this functions?

Reply:
where do u see the problem?

Reply:
Why do we need "have int func to obtain seconds since epoch" if already there is oneliner for this.